docs/resources/offset.md

```diff
@@ -0,0 +1,88 @@
+subcategory: ""
+description: |-
+  Manages an offset time resource, which keeps an UTC timestamp stored in the Terraform state that is offset from a locally sourced base timestamp. This prevents perpetual differences caused by using the timestamp() function https://www.terraform.io/docs/configuration/functions/timestamp.html.
+---
+
+# time_offset (Resource)
+
+Manages an offset time resource, which keeps an UTC timestamp stored in the Terraform state that is offset from a locally sourced base timestamp. This prevents perpetual differences caused by using the [`timestamp()` function](https://www.terraform.io/docs/configuration/functions/timestamp.html).
+
+-> Further manipulation of incoming or outgoing values can be accomplished with the [`formatdate()` function](https://www.terraform.io/docs/configuration/functions/formatdate.html) and the [`timeadd()` function](https://www.terraform.io/docs/configuration/functions/timeadd.html).
+
+## Example Usage
+
+### Basic Usage
+
+```terraform
+resource "time_offset" "example" {
+  offset_days = 7
+}
+
+output "one_week_from_now" {
+  value = time_offset.example.rfc3339
+}
+```
+
+### Triggers Usage
+
+```terraform
+resource "time_offset" "ami_update" {
+  triggers = {
+    # Save the time each switch of an AMI id
+    ami_id = data.aws_ami.example.id
+  }
+
+  offset_days = 7
+}
+
+resource "aws_instance" "server" {
+  # Read the AMI id "through" the time_offset resource to ensure that
+  # both will change together.
+  ami = time_offset.ami_update.triggers.ami_id
+
+  tags = {
+    ExpirationTime = time_offset.ami_update.rfc3339
+  }
+
+  # ... (other aws_instance arguments) ...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `base_rfc3339` (String) Configure the base timestamp with an UTC [RFC3339 time string](https://tools.ietf.org/html/rfc3339#section-5.8) (`YYYY-MM-DDTHH:MM:SSZ`). Defaults to the current time.
+- `offset_days` (Number) Number of days to offset the base timestamp. Conflicts with other `offset_` arguments.
+- `offset_hours` (Number) Number of hours to offset the base timestamp. Conflicts with other `offset_` arguments.
+- `offset_minutes` (Number) Number of minutes to offset the base timestamp. Conflicts with other `offset_` arguments.
+- `offset_months` (Number) Number of months to offset the base timestamp. Conflicts with other `offset_` arguments.
+- `offset_seconds` (Number) Number of seconds to offset the base timestamp. Conflicts with other `offset_` arguments.
+- `offset_years` (Number) Number of years to offset the base timestamp. Conflicts with other `offset_` arguments.
+- `triggers` (Map of String) Arbitrary map of values that, when changed, will trigger a new base timestamp value to be saved. See [the main provider documentation](../index.md) for more information.
+
+### Read-Only
+
+- `day` (Number) Number day of offset timestamp.
+- `hour` (Number) Number hour of offset timestamp.
+- `id` (String) The ID of this resource.
```

Comment:
So, here you are getting the `id` definition "defaulted" because the `tfplugindocs` can only put in the description, what it knows about.

Before the doc was manually maintained, and it said something like:

```
id - UTC RFC3339 format of the base timestamp, e.g. `2020-02-12T06:36:13Z`.
```

Now, to achieve the same goal, you need to add an additional `id` attribute to the schema of each resource. `id` is implicit (all data sources and resources have it), but giving it a `.Description` is optional.

We can talk in zoom about this, but meanwhile here is an example (`id` set in the schema of the `private_key` in TLS provider https://github.com/hashicorp/terraform-provider-tls/blob/main/internal/provider/resource_private_key.go#L138).